Some warnings when building cmsis rtos v2 samples/tests on s32z270dc2

**Describe the bug**
Some warnings when building cmsis rtos v2 samples/tests on s32z270dc2

**To Reproduce**
Steps to reproduce the behavior:
1. Build any of sample/test for cmsis rtos v2, for example:
    `west build -p -b s32z270dc2_rtu0_r52 samples/subsys/portability/cmsis_rtos_v2/philosophers`
2. See the warning

**Expected behavior**
The build is successful without compiler warnings

**Impact**
The CI treats warnings as error so this may block main development

**Logs and console output**
A part of the log is

```
D:/upstream/modules/hal/nxp/s32/soc/s32z27/include/PlatformTypes.h:148: note: this is the location of the previous definition
  148 |     #define TRUE true
      |
D:/upstream/zephyr/subsys/portability/cmsis_rtos_v2/wrapper.h:14: warning: "FALSE" redefined
   14 | #define FALSE   0
      |
D:/upstream/modules/hal/nxp/s32/soc/s32z27/include/PlatformTypes.h:163: note: this is the location of the previous definition
  163 |     #define FALSE false
      |
[42/157] Building C object zephyr/subsys/portability/cmsis_rtos_v2/CMakeFiles/subsys__portability__cmsis_rtos_v2.dir/timer.c.obj
In file included from D:/upstream/zephyr/subsys/portability/cmsis_rtos_v2/timer.c:9:
D:/upstream/zephyr/subsys/portability/cmsis_rtos_v2/wrapper.h:13: warning: "TRUE" redefined
   13 | #define TRUE    1
      |
In file included from D:/upstream/modules/hal/nxp/s32/drivers/s32ze/BaseNXP/include/BasicTypes.h:9,
                 from D:/upstream/modules/hal/nxp/s32/drivers/s32ze/BaseNXP/header/S32Z2_COMMON.h:72,
                 from D:/upstream/modules/hal/nxp/s32/drivers/s32ze/BaseNXP/header/S32Z2_ACCESS_PROTECTION.h:60,
                 from D:/upstream/modules/hal/nxp/s32/drivers/s32ze/BaseNXP/header/S32Z2.h:29,
                 from D:/upstream/zephyr/soc/arm/nxp_s32/s32ze/soc.h:14,
                 from d:\upstream\zephyr\modules\cmsis\cmsis_core_a_r.h:17,
                 from d:\upstream\zephyr\modules\cmsis\cmsis_core.h:12,
                 from D:/upstream/zephyr/include/zephyr/arch/arm/barrier.h:13,
                 from D:/upstream/zephyr/include/zephyr/sys/barrier.h:14,
                 from D:/upstream/zephyr/include/zephyr/arch/arm/cortex_a_r/sys_io.h:19,
                 from D:/upstream/zephyr/include/zephyr/arch/arm/arch.h:42,
                 from D:/upstream/zephyr/include/zephyr/arch/cpu.h:19,
                 from D:/upstream/zephyr/include/zephyr/kernel_includes.h:37,
                 from D:/upstream/zephyr/include/zephyr/kernel.h:17,
                 from D:/upstream/zephyr/subsys/portability/cmsis_rtos_v2/timer.c:7:
D:/upstream/modules/hal/nxp/s32/soc/s32z27/include/PlatformTypes.h:148: note: this is the location of the previous definition
  148 |     #define TRUE true
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: 69d62add980bcafa6d90b73920372390cc10ea7c

--- a/soc/arm/nxp_s32/common/cmsis_rtos_v2_adapt.h
+++ b/soc/arm/nxp_s32/common/cmsis_rtos_v2_adapt.h
@@ -1,0 +1,19 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_SOC_ARM_NXP_S32_COMMON_CMSIS_RTOS_V2_ADAPT_H_
+#define ZEPHYR_SOC_ARM_NXP_S32_COMMON_CMSIS_RTOS_V2_ADAPT_H_
+
+/*
+ * The HAL is defining these symbols already. To avoid interference
+ * between HAL and the CMSIS RTOS wrapper, redefine them under an enum.
+ */
+#undef TRUE
+#undef FALSE
+
+enum { FALSE, TRUE};
+
+#endif /* ZEPHYR_SOC_ARM_NXP_S32_COMMON_CMSIS_RTOS_V2_ADAPT_H_ */

--- a/soc/arm/nxp_s32/s32k3/soc.h
+++ b/soc/arm/nxp_s32/s32k3/soc.h
@@ -10,12 +10,7 @@
 #include <S32K344.h>
 
 #if defined(CONFIG_CMSIS_RTOS_V2)
-/*
- * The HAL is defining these symbols already. To avoid redefinitions,
- * let CMSIS RTOS wrapper define them.
- */
-#undef TRUE
-#undef FALSE
+#include <cmsis_rtos_v2_adapt.h>
 #endif
 
 /* Aliases for peripheral base addresses */

--- a/soc/arm/nxp_s32/s32ze/soc.h
+++ b/soc/arm/nxp_s32/s32ze/soc.h
@@ -16,6 +16,10 @@
 #error "SoC not supported"
 #endif
 
+#if defined(CONFIG_CMSIS_RTOS_V2)
+#include <cmsis_rtos_v2_adapt.h>
+#endif
+
 /* Aliases for peripheral base addresses */
 
 /* SIUL2 */

